### DISCUSSION TITLE
Sending IPv4 raw packet

### DISCUSSION THREAD

**Discussion by eike0** [xjasonlyu/tun2socks#347] [General]
Hi, I'm trying to inject IPv4 packets to a tun2socks interface, i tried forking tun2socks library and using `stack.WriteRawPacket` but can't make it work. 
I can't find any gvisor documentation or example. does anyone has any suggestion or link to an example or documentation on how to do this? any help would be appreciated.

**Comment by xjasonlyu** [comment]
why not just write to the tun interface?

**Reply by eike0** [reply]
can you link any example or documentation on how to do this in windows

**Comment by eike0** [comment]
I don't know how. i'm pretty new to networking, I want to intercept packets from other interfaces and inject it to tun2socks without using routing table, because i want to send traffic for only some applications. I figured out the intercepting part for windows.

can i read raw packets from tun2socks?
